oasislmf/model_preparation/summaries.py

```diff
@@ -413,3 +419,153 @@ def generate_summaryxref_files(model_run_fp, analysis_settings):
+    :type target_dir: str
+
+    :param gul_inputs_df: dataframe from gul_inputs.get_gul_input_items(..)
+    :type gul_inputs_df: pandas.DataFrame
+
+    :param exposure_fp: file path to input exposure file
+    :type exposure_fp: str
+
+    :param keys_errors_fp: file path to keys erors file
+    :type keys_errors_fp: str
+
+    :param exposure_profile: profile defining exposure file
+    :type exposure_profile: dict
+
+    :param loc_num: location number column heading from exposure file
+    :type loc_num: str
+    """
+
+    # Get keys in GUL input items dataframe and merge with key errors to obtain
+    # status information
+    try:
+        gul_inputs_errors_df, _ = get_gul_input_items(
+            exposure_fp, keys_errors_fp, exposure_profile=exposure_profile
+        )
+    except OasisException:   # Empty dataframe
+        gul_inputs_errors_df = pd.DataFrame(columns=gul_inputs_df.columns)
+    keys_errors_df = pd.read_csv(keys_errors_fp)
+    keys_errors_df.columns = keys_errors_df.columns.str.lower()
+    gul_inputs_errors_df = gul_inputs_errors_df.merge(
+        keys_errors_df,
+        left_on=[loc_num, 'peril_id', 'coverage_type_id'],
+        right_on=['locid', 'perilid', 'coveragetypeid'],
+        how='left'
+    )
+    gul_inputs_errors_df.drop(
+        columns=['locid', 'perilid', 'coveragetypeid', 'message'],
+        axis=1,
+        inplace=True
+    )
+
+    # Compile summary of exposure data
+    exposure_summary = {}
+    for peril_code in gul_inputs_df['peril_id'].drop_duplicates():
+        # Use descriptive names of perils as keys
+        for k, vals in PERILS.items():
+            if vals['id'] == peril_code:
+                peril = vals['desc']
+                break
+        exposure_summary[peril] = {}
+        # Create dictionary structure for all and each validity status
+        for status in ['all'] + list(OASIS_KEYS_STATUS.keys()):
+            exposure_summary[peril][status] = {}
+            exposure_summary[peril][status]['tiv'] = 0.0
+            exposure_summary[peril][status]['tiv_by_coverage'] = {}
+            exposure_summary[peril][status]['number_of_locations'] = 0
+            # Fill exposure summary dictionary
+            if status == 'success':
+                exposure_summary = get_exposure_summary(
+                    gul_inputs_df,
+                    exposure_summary,
+                    peril,
+                    peril_code,
+                    status,
+                    loc_num
+                )
+            elif status != 'all':
+                exposure_summary = get_exposure_summary(
+                    gul_inputs_errors_df[gul_inputs_errors_df['status']==status],
+                    exposure_summary,
+                    peril,
+                    peril_code,
+                    status,
+                    loc_num
+                )
+
+    # Write exposure summary as json file
+    with io.open(
```

Comment:
Pandas provides a `to_json` method for writing a dataframe to a JSON file. That might be more convenient. For example, we use the `to_csv` method all the time to write CSV files from dataframes, which is much more convenient than explicitly doing the file IO bit yourself.